src/hub/Hub.sol

```diff
@@ -380,6 +380,39 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler, IHubGuar
         sender.sendRevokedShares(poolId, scId, payoutAssetId, payoutAssetAmount, revokedShareAmount, navPoolPerShare);
     }
 
+    /// @inheritdoc IHub
+    function forceCancelDepositRequest(PoolId poolId, ShareClassId scId, bytes32 investor, AssetId depositAssetId)
+        external
+        payable
+        payTransaction
+    {
+        _isManager(poolId);
+
+        uint128 cancelledAssetAmount =
+            shareClassManager.forceCancelDepositRequest(poolId, scId, investor, depositAssetId);
+
+        // Cancellation might have been queued such that it will be executed in the future during claiming
+        if (cancelledAssetAmount > 0) {
+            sender.sendFulfilledDepositRequest(poolId, scId, depositAssetId, investor, 0, 0, cancelledAssetAmount);
```

Comment:
Nice to have one message for fulfilling and cancelling. "One message to rule them all"